File: DomainSpecificLanguages/DomainSpecificLanguages/AlarmsWithNestedFunctions/Alarms/CompoundAlarm.cs
namespace ChangingBehaviorThroughComposition
{
  public class CompoundAlarm : Alarm
  {
    private readonly Alarm[] _alarms;
  
    public CompoundAlarm(params Alarm[] alarms)
    {
      _alarms = alarms;
    }

    public void Trigger()
    {
      foreach (var alarm in _alarms)
      {
        alarm.Trigger();
      }
    }

    public void Disable()
    {
      foreach (var alarm in _alarms)
      {
        alarm.Disable();
      }
    }
  }
}

File: Java/dependency-injection/changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/Alarms/CompoundAlarm.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.Alarms;

import com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.Alarm;

public class CompoundAlarm implements Alarm {
  private final Alarm[] _alarms;

  public CompoundAlarm(Alarm... alarms) {
    _alarms = alarms;
  }

  public void trigger() {
    for (Alarm alarm : _alarms) {
      alarm.trigger();
    }
  }

  public void disable() {
    for (Alarm alarm : _alarms) {
      alarm.disable();
    }
  }
}
